sdk/provisioning/Azure.Provisioning.CloudMachine/src/OFX/StorageFile.cs

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.ComponentModel;
+using System.Threading;
+
+namespace Azure.CloudMachine;
+
+public class StorageFile
+{
+    private readonly Response? _response;
+
+    private StorageServices _storage;
+    public string Path { get; internal set; }
+
+    /// <summary>
+    /// The requestId for the storage operation that triggered this event
+    /// </summary>
+    public string RequestId { get; internal set; }
+
+    /// <summary>
+    ///
+    /// </summary>
+    /// <param name="result"></param>
+    /// <remarks>returns null if the file is not created as a return value of a service method call.</remarks>
+    public static implicit operator Response?(StorageFile result) => result._response;
+
+    public CancellationToken CancellationToken { get; internal set; }
+
+    public BinaryData Download()
```

Comment:
Should this be async? My naive impression was most things in CM were async by default.

Reply:
will consider this for later